apps/aecore/lib/aecore/channel/worker.ex

```diff
@@ -511,6 +516,25 @@ defmodule Aecore.Channel.Worker do
     end
   end
 
+  @doc """
+    Submits a snapshot of the most recent state of a channel
+  """
+  def handle_call({:snapshot, channel_id, fee, nonce, priv_key}, _from, state) do
+    peer_state = Map.get(state, channel_id)
+
+    with {:ok, tx} <-
+      ChannelStatePeer.snapshot(peer_state, fee, nonce, priv_key),
+         :ok <- Pool.add_transaction(tx) do
+        {:reply, :ok, state}
+      else
+      {:error, _} = err ->
+        {:reply, err, state}
+
+      :error ->
+        {:error, "#{__MODULE__}: Pool error"}
```

Comment:
why not reply here as well? also now that I see that Pool.add_transaction replies with error and no message which requires you to do 2 different checks just for errors, could you make it reply with {:error, message}